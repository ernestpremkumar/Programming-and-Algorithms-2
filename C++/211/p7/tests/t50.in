insert 2
find 1
find 2
breadth
echo 3
print
insert 6
find 3
echo 7
distance
breadth
distance
print
print
size
distance
insert 3
breadth
print
balanced
find 4
find 3
balanced
find 6
distance
breadth
breadth
size
echo 1
breadth
distance
distance
breadth
find 3
distance
breadth
distance
breadth
balanced
print
size
insert 1
breadth
insert 3
size
echo 6
size
echo 6
insert 7
echo 4
distance
echo 5
distance
insert 5
distance
size
balanced
echo 3
balanced
size
balanced
size
echo 1
echo 3
balanced
insert 0
distance
size
balanced
insert 3
insert 5
breadth
balanced
print
distance
balanced
echo 6
balanced
breadth
size
echo 4
insert 3
print
size
balanced
echo 3
balanced
echo 6
find 6
print
find 2
echo 3
breadth
print
balanced
distance
insert 0
distance
find 5
insert 1
print
size
balanced
insert 2
balanced
breadth
insert 2
breadth
print
insert 1
balanced
print
balanced
distance
find 3
breadth
echo 0
find 0
size
find 5
size
find 0
breadth
size
size
breadth
size
size
print
breadth
balanced
balanced
size
echo 3
echo 5
distance
find 2
distance
find 0
insert 0
find 5
balanced
echo 5
balanced
size
print
balanced
size
distance
distance
breadth
insert 5
balanced
echo 7
balanced
balanced
echo 2
balanced
insert 1
breadth
find 3
breadth
breadth
print
insert 3
size
balanced
distance
size
echo 6
echo 1
find 0
balanced
insert 2
insert 0
find 7
echo 5
balanced
echo 5
insert 3
distance
echo 3
distance
print
breadth
insert 4
print
distance
distance
find 7
distance
breadth
echo 7
print
echo 0
size
echo 6
find 3
size
distance
print
breadth
find 0
print
size
size
distance
echo 7
find 6
balanced
size
insert 1
print
insert 5
print
distance
balanced
insert 3
distance
print
print
insert 6
breadth
insert 1
distance
print
size
size
find 6
breadth
find 7
echo 3
distance
print
distance
print
balanced
breadth
balanced
distance
balanced
breadth
echo 1
echo 0
find 3
find 5
echo 6
echo 3
print
size
distance
print
print
size
insert 4
insert 0
find 0
size
balanced
balanced
balanced
insert 2
breadth
breadth
distance
find 7
breadth
size
insert 2
distance
breadth
find 7
distance
find 5
balanced
echo 2
print
breadth
size
size
insert 1
breadth
breadth
echo 7
print
insert 6
distance
insert 4
insert 4
size
balanced
distance
find 3
breadth
breadth
distance
breadth
print
find 2
echo 5
find 7
echo 1
breadth
distance
distance
distance
size
echo 4
balanced
size
breadth
size
breadth
print
print
echo 4
distance
insert 0
size
size
balanced
breadth
breadth
distance
find 4
insert 0
insert 0
size
size
balanced
size
distance
balanced
print
distance
insert 6
distance
distance
print
echo 4
print
balanced
print
print
balanced
breadth
size
size
size
find 6
print
breadth
find 2
distance
balanced
distance
balanced
size
distance
size
size
balanced
find 4
distance
balanced
distance
balanced
breadth
balanced
size
insert 7
find 0
distance
print
find 4
size
distance
size
size
echo 0
size
print
size
print
breadth
insert 6
distance
balanced
insert 4
insert 3
insert 4
size
print
size
print
balanced
echo 7
echo 3
print
breadth
size
echo 6
size
breadth
size
balanced
insert 3
find 1
breadth
find 5
balanced
insert 2
breadth
find 1
distance
find 0
distance
insert 3
size
find 7
breadth
echo 2
print
print
insert 1
find 0
size
echo 4
distance
size
find 4
print
insert 1
insert 6
breadth
echo 4
balanced
insert 0
find 3
print
distance
breadth
print
echo 5
insert 2
distance
echo 4
distance
insert 1
print
print
echo 4
print
distance
echo 0
print
size
distance
balanced
insert 7
insert 0
print
find 1
size
size
size
find 6
insert 2
print
echo 7
insert 4
size
distance
find 1
echo 5
breadth
print
find 1
print
find 5
print
print
echo 0
echo 5
print
insert 0
print
insert 3
print
balanced
insert 4
distance
distance
breadth
distance
echo 0
distance
find 4
insert 6
size
find 5
size
balanced
size
echo 3
echo 1
distance
balanced
balanced
distance
print
echo 6
size
echo 3
distance
echo 3
distance
insert 2
distance
insert 3
print
echo 1
balanced
distance
find 6
find 4
size
size
balanced
balanced
distance
balanced
size
size
find 0
balanced
print
size
balanced
insert 1
breadth
find 4
echo 0
print
echo 0
distance
find 6
find 5
distance
distance
echo 1
insert 5
breadth
balanced
breadth
echo 0
print
size
echo 5
breadth
find 1
balanced
echo 6
print
balanced
echo 2
balanced
find 1
breadth
size
size
breadth
breadth
print
distance
distance
print
echo 6
echo 4
find 4
distance
balanced
print
size
breadth
print
distance
balanced
insert 3
insert 3
print
insert 7
breadth
print
breadth
distance
size
echo 5
echo 7
insert 3
balanced
distance
balanced
insert 5
echo 0
balanced
size
print
insert 7
breadth
echo 6
insert 3
insert 2
insert 2
breadth
balanced
echo 4
size
size
balanced
distance
distance
balanced
echo 2
find 0
balanced
insert 3
breadth
size
find 7
insert 7
print
echo 1
echo 6
size
breadth
distance
insert 4
distance
breadth
balanced
print
breadth
size
insert 7
balanced
echo 2
print
insert 1
print
print
breadth
breadth
balanced
echo 4
breadth
insert 7
balanced
print
breadth
distance
balanced
balanced
echo 3
echo 1
breadth
breadth
distance
echo 3
balanced
insert 0
balanced
distance
echo 6
balanced
balanced
size
breadth
size
find 5
insert 2
distance
size
insert 4
insert 2
find 0
breadth
breadth
insert 3
distance
print
distance
print
echo 3
echo 7
insert 3
find 1
echo 0
print
echo 1
size
insert 3
breadth
size
print
balanced
distance
balanced
insert 2
insert 7
find 1
find 3
find 6
print
print
breadth
balanced
breadth
breadth
find 0
echo 1
size
balanced
insert 1
echo 0
size
breadth
size
echo 0
echo 6
size
find 5
print
breadth
echo 1
size
size
balanced
breadth
size
find 6
insert 4
insert 0
insert 3
insert 1
distance
distance
print
insert 2
find 2
size
find 6
size
balanced
find 0
distance
distance
distance
find 7
insert 7
echo 4
echo 7
echo 7
insert 4
distance
find 5
find 1
distance
insert 5
breadth
distance
size
find 4
print
echo 6
size
print
balanced
balanced
breadth
distance
distance
balanced
distance
insert 0
print
balanced
print
balanced
echo 0
breadth
balanced
breadth
breadth
echo 4
size
print
balanced
breadth
print
insert 6
distance
balanced
find 3
find 6
breadth
balanced
size
find 4
insert 0
find 3
insert 3
find 7
size
distance
balanced
balanced
echo 5
echo 6
print
echo 4
breadth
breadth
find 7
balanced
print
balanced
insert 6
distance
find 4
size
print
echo 1
balanced
echo 0
print
print
balanced
insert 2
find 3
balanced
find 7
breadth
size
echo 4
echo 5
echo 3
print
insert 0
echo 4
balanced
size
distance
distance
print
echo 1
echo 4
find 3
balanced
size
size
find 2
balanced
size
find 0
insert 3
echo 3
balanced
breadth
distance
distance
breadth
insert 6
print
breadth
echo 6
insert 0
size
find 0
echo 5
echo 6
insert 1
breadth
distance
breadth
find 4
size
size
insert 3
print
breadth
balanced
echo 6
distance
distance
insert 6
balanced
distance
find 4
distance
insert 1
print
find 4
balanced
breadth
find 3
echo 2
echo 2
print
find 6
print
insert 4
balanced
echo 7
print
print
echo 6
size
balanced
echo 1
breadth
distance
distance
breadth
print
print
print
find 5
breadth
balanced
size
balanced
find 2
distance
find 2
distance
insert 0
breadth
size
insert 2
size
distance
find 6
size
breadth
size
print
size
find 0
size
distance
distance
breadth
balanced
breadth
distance
distance
breadth
balanced
size
insert 4
print
distance
distance
distance
balanced
insert 4
find 5
echo 7
balanced
find 5
print
size
find 1
insert 1
distance
size
insert 2
echo 0
insert 7
balanced
size
breadth
echo 5
echo 5
size
print
print
echo 7
echo 0
distance
print
echo 0
insert 5
print
distance
find 4
breadth
distance
distance
