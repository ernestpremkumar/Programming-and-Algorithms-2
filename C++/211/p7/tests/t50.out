<1> is not in tree.
<2> is in tree.
{2}
3
{2}
<3> is not in tree.
7
Average distance of nodes to root = 0.5
{2, 6}
Average distance of nodes to root = 0.5
{2, 6}
{2, 6}
2
Average distance of nodes to root = 0.5
{2, 6, 3}
{2, 3, 6}
Tree is not balanced.
<4> is not in tree.
<3> is in tree.
Tree is not balanced.
<6> is in tree.
Average distance of nodes to root = 1
{2, 6, 3}
{2, 6, 3}
3
1
{2, 6, 3}
Average distance of nodes to root = 1
Average distance of nodes to root = 1
{2, 6, 3}
<3> is in tree.
Average distance of nodes to root = 1
{2, 6, 3}
Average distance of nodes to root = 1
{2, 6, 3}
Tree is not balanced.
{2, 3, 6}
3
{2, 1, 6, 3}
4
6
4
6
4
Average distance of nodes to root = 1.2
5
Average distance of nodes to root = 1.2
Average distance of nodes to root = 1.5
6
Tree is not balanced.
3
Tree is not balanced.
6
Tree is not balanced.
6
1
3
Tree is not balanced.
Average distance of nodes to root = 1.57143
7
Tree is balanced.
{2, 1, 6, 0, 3, 7, 5}
Tree is balanced.
{0, 1, 2, 3, 5, 6, 7}
Average distance of nodes to root = 1.57143
Tree is balanced.
6
Tree is balanced.
{2, 1, 6, 0, 3, 7, 5}
7
4
{0, 1, 2, 3, 5, 6, 7}
7
Tree is balanced.
3
Tree is balanced.
6
<6> is in tree.
{0, 1, 2, 3, 5, 6, 7}
<2> is in tree.
3
{2, 1, 6, 0, 3, 7, 5}
{0, 1, 2, 3, 5, 6, 7}
Tree is balanced.
Average distance of nodes to root = 1.57143
Average distance of nodes to root = 1.57143
<5> is in tree.
{0, 1, 2, 3, 5, 6, 7}
7
Tree is balanced.
Tree is balanced.
{2, 1, 6, 0, 3, 7, 5}
{2, 1, 6, 0, 3, 7, 5}
{0, 1, 2, 3, 5, 6, 7}
Tree is balanced.
{0, 1, 2, 3, 5, 6, 7}
Tree is balanced.
Average distance of nodes to root = 1.57143
<3> is in tree.
{2, 1, 6, 0, 3, 7, 5}
0
<0> is in tree.
7
<5> is in tree.
7
<0> is in tree.
{2, 1, 6, 0, 3, 7, 5}
7
7
{2, 1, 6, 0, 3, 7, 5}
7
7
{0, 1, 2, 3, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5}
Tree is balanced.
Tree is balanced.
7
3
5
Average distance of nodes to root = 1.57143
<2> is in tree.
Average distance of nodes to root = 1.57143
<0> is in tree.
<5> is in tree.
Tree is balanced.
5
Tree is balanced.
7
{0, 1, 2, 3, 5, 6, 7}
Tree is balanced.
7
Average distance of nodes to root = 1.57143
Average distance of nodes to root = 1.57143
{2, 1, 6, 0, 3, 7, 5}
Tree is balanced.
7
Tree is balanced.
Tree is balanced.
2
Tree is balanced.
{2, 1, 6, 0, 3, 7, 5}
<3> is in tree.
{2, 1, 6, 0, 3, 7, 5}
{2, 1, 6, 0, 3, 7, 5}
{0, 1, 2, 3, 5, 6, 7}
7
Tree is balanced.
Average distance of nodes to root = 1.57143
7
6
1
<0> is in tree.
Tree is balanced.
<7> is in tree.
5
Tree is balanced.
5
Average distance of nodes to root = 1.57143
3
Average distance of nodes to root = 1.57143
{0, 1, 2, 3, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<7> is in tree.
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
7
{0, 1, 2, 3, 4, 5, 6, 7}
0
8
6
<3> is in tree.
8
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
Average distance of nodes to root = 1.875
7
<6> is in tree.
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Tree is not balanced.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
<6> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
<7> is in tree.
3
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
1
0
<3> is in tree.
<5> is in tree.
6
3
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
<0> is in tree.
8
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
<7> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
8
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
<7> is in tree.
Average distance of nodes to root = 1.875
<5> is in tree.
Tree is not balanced.
2
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
8
8
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
7
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
8
Tree is not balanced.
Average distance of nodes to root = 1.875
<3> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
5
<7> is in tree.
1
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
8
4
Tree is not balanced.
8
{2, 1, 6, 0, 3, 7, 5, 4}
8
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
4
Average distance of nodes to root = 1.875
8
8
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
<4> is in tree.
8
8
Tree is not balanced.
8
Average distance of nodes to root = 1.875
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
4
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
8
8
8
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
<2> is in tree.
Average distance of nodes to root = 1.875
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
8
Average distance of nodes to root = 1.875
8
8
Tree is not balanced.
<4> is in tree.
Average distance of nodes to root = 1.875
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
8
<0> is in tree.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
8
Average distance of nodes to root = 1.875
8
8
0
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
7
3
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
8
6
8
{2, 1, 6, 0, 3, 7, 5, 4}
8
Tree is not balanced.
<1> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
<5> is in tree.
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
<1> is in tree.
Average distance of nodes to root = 1.875
<0> is in tree.
Average distance of nodes to root = 1.875
8
<7> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
2
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
8
4
Average distance of nodes to root = 1.875
8
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
4
Tree is not balanced.
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
5
Average distance of nodes to root = 1.875
4
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
4
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
0
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 1.875
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
8
8
8
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
7
8
Average distance of nodes to root = 1.875
<1> is in tree.
5
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
0
5
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
0
Average distance of nodes to root = 1.875
<4> is in tree.
8
<5> is in tree.
8
Tree is not balanced.
8
3
1
Average distance of nodes to root = 1.875
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
6
8
3
Average distance of nodes to root = 1.875
3
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
1
Tree is not balanced.
Average distance of nodes to root = 1.875
<6> is in tree.
<4> is in tree.
8
8
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
8
8
<0> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
<4> is in tree.
0
{0, 1, 2, 3, 4, 5, 6, 7}
0
Average distance of nodes to root = 1.875
<6> is in tree.
<5> is in tree.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
1
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
0
{0, 1, 2, 3, 4, 5, 6, 7}
8
5
{2, 1, 6, 0, 3, 7, 5, 4}
<1> is in tree.
Tree is not balanced.
6
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
2
Tree is not balanced.
<1> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
8
8
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
6
4
<4> is in tree.
Average distance of nodes to root = 1.875
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
8
5
7
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
0
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
6
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
4
8
8
Tree is not balanced.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Tree is not balanced.
2
<0> is in tree.
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
8
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
1
6
8
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
8
Tree is not balanced.
2
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
4
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Tree is not balanced.
Tree is not balanced.
3
1
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
3
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 1.875
6
Tree is not balanced.
Tree is not balanced.
8
{2, 1, 6, 0, 3, 7, 5, 4}
8
<5> is in tree.
Average distance of nodes to root = 1.875
8
<0> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
3
7
<1> is in tree.
0
{0, 1, 2, 3, 4, 5, 6, 7}
1
8
{2, 1, 6, 0, 3, 7, 5, 4}
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
<1> is in tree.
<3> is in tree.
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
<0> is in tree.
1
8
Tree is not balanced.
0
8
{2, 1, 6, 0, 3, 7, 5, 4}
8
0
6
8
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
1
8
8
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
8
<6> is in tree.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
8
<6> is in tree.
8
Tree is not balanced.
<0> is in tree.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<7> is in tree.
4
7
7
Average distance of nodes to root = 1.875
<5> is in tree.
<1> is in tree.
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
8
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
6
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Tree is not balanced.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
0
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
4
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Tree is not balanced.
<3> is in tree.
<6> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
8
<4> is in tree.
<3> is in tree.
<7> is in tree.
8
Average distance of nodes to root = 1.875
Tree is not balanced.
Tree is not balanced.
5
6
{0, 1, 2, 3, 4, 5, 6, 7}
4
{2, 1, 6, 0, 3, 7, 5, 4}
{2, 1, 6, 0, 3, 7, 5, 4}
<7> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 1.875
<4> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
1
Tree is not balanced.
0
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<3> is in tree.
Tree is not balanced.
<7> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
8
4
5
3
{0, 1, 2, 3, 4, 5, 6, 7}
4
Tree is not balanced.
8
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
1
4
<3> is in tree.
Tree is not balanced.
8
8
<2> is in tree.
Tree is not balanced.
8
<0> is in tree.
3
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
6
8
<0> is in tree.
5
6
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
<4> is in tree.
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
6
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Tree is not balanced.
Average distance of nodes to root = 1.875
<4> is in tree.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
<3> is in tree.
2
2
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
7
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
6
8
Tree is not balanced.
1
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
8
Tree is not balanced.
<2> is in tree.
Average distance of nodes to root = 1.875
<2> is in tree.
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
8
8
Average distance of nodes to root = 1.875
<6> is in tree.
8
{2, 1, 6, 0, 3, 7, 5, 4}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
<0> is in tree.
8
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{2, 1, 6, 0, 3, 7, 5, 4}
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Tree is not balanced.
<5> is in tree.
7
Tree is not balanced.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
<1> is in tree.
Average distance of nodes to root = 1.875
8
0
Tree is not balanced.
8
{2, 1, 6, 0, 3, 7, 5, 4}
5
5
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
7
0
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
0
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
<4> is in tree.
{2, 1, 6, 0, 3, 7, 5, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
